src/lang.ts

```diff
@@ -213,3 +214,20 @@ export function partial(targetFunction: (...args: any[]) => any, ...suppliedArgs
 		return targetFunction.apply(this, args);
 	};
 }
+
+export function createHandle(destructor: () => void): Handle {
+	return {
+		destroy: function () {
+			this.destroy = function () {};
+			destructor.call(this);
+		}
+	};
+}
+
+export function createCompositeHandle(...handles: Handle[]): Handle {
+	return createHandle(function () {
+		for (var i = 0, handle: Handle; (handle = handles[i]); ++i) {
```

Comment:
`for...of` compiles to a `for` loop when targeting ES3 and ES5 ([reference](https://github.com/Microsoft/TypeScript/wiki/What%27s-new-in-TypeScript#forof-support))